### ISSUE TITLE
paths on windows?

### ISSUE THREAD

**Issue by kaelzhang** [kaelzhang/node-ignore#5]

**Comment by zxqfox** [comment]
Something not work on win?

**Comment by PhiLhoSoft** [comment]
Git on Windows: `git status` reports `foo/bar` for `foo\bar`, so I use `foo/bar` syntax in `.gitignore`, and it just works.
Note that in most commands / use cases, using forward slashes in Windows just works.
If _ignore_ is given a classical Windows path like `src\main\java`, it won't match.
So, currently, it is the responsibility of the caller to convert these slashes.
It doesn't shock me, as the caller also must convert absolute paths to paths relative to the repository, and must ensure folder paths end with a slash.

**Comment by kaelzhang** [comment]
First of all, I thought it is better to use `foo/bar` in .gitignore even on Windows.

Maybe I could convert path like `path\to\` into `path/to` before testing on Windows.

**Comment by IanVS** [comment]
Are Windows paths fully supported after https://github.com/kaelzhang/node-ignore/pull/9?  If so, can this issue be closed?

**Comment by gronke** [comment]
All unit tests pass on Windows 10 with Node 5.7.1 ([b0dd0858f90c39b4b923f3a0fbeae222d375faa7](https://github.com/kaelzhang/node-ignore/tree/b0dd0858f90c39b4b923f3a0fbeae222d375faa7))

**Comment by IanVS** [comment]
I did an experiment by using the examples in the readme but normalizing the paths prior to applying the filter.

```js
var ignore = require('ignore');
var path = require('path');

var ig = ignore().addPattern(['.abc/*', '!.abc/d/']);

var paths = [
  path.normalize('.abc/a.js'),    // should be filtered out
  path.normalize('.abc/d/e.js')   // should be included
]
console.log('paths:', paths);

console.log('filtered:', ig.filter(paths));  // ['.abc/d/e.js']
```

On Mac, the result (with version `2.2.19`, as `3.0.3` has other issues, namely #14) is:

```js
paths:    [ '.abc/a.js', '.abc/d/e.js' ]
filtered: [ '.abc/d/e.js' ]
```

But on Windows (with versions `2.2.19` and `3.0.2`, I get:

```js
paths:    [ '.abc\\a.js', '.abc\\d\\e.js' ]
filtered: [ '.abc\\a.js', '.abc\\d\\e.js' ]
```

So, how should Windows paths be handled when using this library?  Do we need to manually convert them to posix paths?

**Comment by kaelzhang** [comment]
Yeah, I need to convert windows paths into linux paths before matching. this will be fixed in `3.0.x`
